### PR TITLE
[ENH] ClaSP: Adds parallelization for distance computations and numbarize function calls

### PR DESCRIPTION
**Pull request by patrickzib**
This is mostly a performance enhancement of the ClaSP implementation by:

- Parallelizing ClaSP distance computations
- Numbarizing methods.
- exposing n_jobs to ClaSP-Constructor class
- convert ClaSPTransformer to BaseSeriesTransformer

### CODE REVIEW

**Comment by TonyBagnall** [comment]
maybe convert to a BaseSeriesTransformer at the same time (#1618)?

**Comment by patrickzib** [comment]
I was thinking about removing the Transformer all together? :)  Or should I convert it?

**Comment by patrickzib** [comment]
> I was thinking about removing the Transformer all together and merge it into ClaSP Segmenter? :) Or should I convert it?

I meant merge the Transformer with the Segmenter, and remove the Transformer

File: aeon/transformations/clasp.py [hunk 1]
```diff
@@ -403,11 +445,16 @@ class ClaSPTransformer(BaseTransformer):
     }
 
     def __init__(
-        self, window_length=10, scoring_metric="ROC_AUC", exclusion_radius=0.05
+        self,
+        window_length=10,
+        scoring_metric="ROC_AUC",
+        exclusion_radius=0.05,
+        n_jobs=4,
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/clasp.py:452]: I would default to 1. Think i will eventually add a test to enforce this.

File: aeon/transformations/clasp.py [hunk 2]
```diff
@@ -316,6 +353,7 @@ def clasp(
     score=_roc_auc_score,
     interpolate=True,
     exclusion_radius=0.05,
+    n_jobs=4,
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/clasp.py:356]: same as other comment

File: aeon/transformations/clasp.py [hunk 3]
```diff
@@ -60,20 +88,20 @@ def _sliding_mean_std(X, m):
...
-    movstd = np.where(abs(movstd) < 0.001, 1, movstd)
-
+    movstd = np.sqrt(np.clip(segSumSq / m - (segSum / m) ** 2, 0, None))
+    movstd = np.where(np.abs(movstd) < 0.001, 1, movstd)
     return [movmean, movstd]
 
 
-def _compute_distances_iterative(X, m, k):
+@njit(fastmath=True, cache=True, parallel=True)
+def _compute_distances_iterative(X, m, k, n_jobs=4, slack=0.5):
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/clasp.py:104]: here as well, but its private so does not really matter i guess.

**Comment by TonyBagnall** [comment]
> I was thinking about removing the Transformer all together and merge it into ClaSP Segmenter? :) Or should I convert it?

I think remove the transformer and just merge it into a clasp segmenter is best way forward

**Comment by patrickzib** [comment]
Thank you @MatthewMiddlehurst and @TonyBagnall 


I set n_jobs to 1 and have converted the BaseTransformer to a BaseSeriesTransformer for now. 

Felt like merging everything into one module is also not the best option for now. Might re-consider this in the future :)

**Review by MatthewMiddlehurst — APPROVED** [review]
LGTM

**Pull request merged**